db/migrate/20150930183738_migrate_content_hosts.rb

```diff
@@ -0,0 +1,185 @@
+  end
+
+  def logger
+    Rails.logger
+  end
+
+  def create_content_aspect(host, system)
+    logger.info("Creating content aspect for host #{host.name}.")
+    host.content_aspect = ::Katello::Host::ContentAspect.new(:content_view => system.content_view, :lifecycle_environment => system.environment)
+  end
+
+  def create_subscription_aspect(host, system)
+    logger.info("Creating subscription aspect for host #{host.name}.")
+    subscription_aspect = host.subscription_aspect = Katello::Host::SubscriptionAspect.new
+    subscription_aspect.activation_keys = system.activation_keys
+  end
+
+  def get_systems_with_facts(systems)
+    systems_to_remove = []
+
+    systems.each do |system|
+      begin
+        facts = system.facts
+        unless facts
+          systems_to_remove.push(system)
+        end
+      rescue RestClient::Exception
+        systems_to_remove.push(system)
+      end
+    end
+
+    systems_to_remove.each do |system|
+      logger.info("Content Host #{system.uuid} doesn't have candlepin information, unregistering.")
+      systems.delete(system)
+      unregister_system(system)
+    end
+
+    systems
+  end
+
+  def group_systems_by_hostname(systems)
+    system_hostnames = {}
+
+    systems.each do |system|
+      hostname = system.facts['network.hostname']
+
+      if system_hostnames[hostname]
+        system_hostnames[hostname].push(system)
+      else
+        system_hostnames[hostname] = [system]
+      end
+    end
+
+    system_hostnames
+  end
+
+  def unregister_system(system)
+    logger.warn("Unregistering content host with UUID: #{system.uuid}")
+
+    begin
+      logger.info("Removing Candlepin consumer #{system.uuid}")
+      Katello::Resources::Candlepin::Consumer.destroy(system.uuid)
+    rescue RestClient::Exception => e
+      logger.warn("Exception when destroying candlepin consumer #{system.uuid}:#{e.inspect}")
+    end
+
+    begin
+      logger.info("Removing Pulp consumer #{system.uuid}")
+      Katello.pulp_server.extensions.consumer.delete(system.uuid)
+    rescue RestClient::ResourceNotFound
+      logger.warn("Pulp consumer not found for consumer #{system.uui} proceeding.")
+      #do nothing
+    rescue RestClient::Exception => e
+      logger.warn("Exception when destroying pulp consumer #{system.uuid}:#{e.inspect}")
+    end
+
+    logger.info("Removing system #{system.uuid}")
+    system.destroy
+  end
+
+  def unregister_all_but_last_system(systems)
+    systems_by_created_date = systems.sort_by(&:created_at)
+    system = systems_by_created_date.pop
+
+    systems_by_created_date.each do |system_to_remove|
+      unregister_system(system_to_remove)
+    end
+
+    system
+  end
+
+  # rubocop:disable MethodLength
+  def up
+    begin
+      User.current = User.anonymous_api_admin
+    rescue Foreman::Exception
+      logger.warn("Foreman anonymous admin does not exist, skipping content host migration.")
+      return
+    end
```

Comment:
I think i prefer an explicit check and return for that user rather than relying on the exception.  Something like:

if User.where(:login => User::ANONYMOUS_API_ADMIN).first.nil?
  logger.warn()
  return
end